BASE/src/ServerTelemetryChannel/Implementation/TransmissionSender.cs

```diff
@@ -134,6 +140,41 @@ public virtual bool Enqueue(Func<Transmission> transmissionGetter)
             return enqueueSucceded;
         }
 
+        /// <summary>
+        /// Wait for inflight transmissions to complete. Honors the CancellationToken set by an application on IAsyncFlushable.FlushAsync.
+        /// </summary>
+        internal Task<TaskStatus> WaitForPreviousTransmissionsToComplete(CancellationToken cancellationToken)
+        {
+            var transmissionFlushAsyncId = this.inFlightTransmissions.LastOrDefault().Key;
```

Comment:
what exactly does LastOrDefault does here? Gets the transmission with highest id? or the one which was added most recent?

Reply:
The one that got added most recent.
Most of the times, recent transmission should be the one with highest id.